core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -227,21 +233,7 @@ trait WhiskActionsApi extends WhiskCollectionAPI with SequenceActions with Refer
                             case Success(true) =>
                                 transid.started(this, if (blocking) LoggingMarkers.CONTROLLER_ACTIVATION_BLOCKING else LoggingMarkers.CONTROLLER_ACTIVATION)
 
-                                val postToLoadBalancer = {
-                                    action.exec match {
-                                        // this is a topmost sequence
-                                        case SequenceExec(_, components) =>
-                                            val futureSeqTuple = invokeSequence(user, action, payload, blocking, topmost = true, components, cause = None, 0)
-                                            futureSeqTuple map { case (activationId, wskActivation, _) => (activationId, wskActivation) }
-                                        case _ => {
-                                            val duration = action.limits.timeout()
-                                            val timeout = (maxWaitForBlockingActivation min duration) + blockingInvokeGrace
-                                            invokeSingleAction(user, action, env, payload, timeout, blocking)
-                                        }
-                                    }
-                                }
-
-                                onComplete(postToLoadBalancer) {
+                                onComplete(invokeAction(user, env map { action.inherit(_) } getOrElse action, payload, blocking, waitOverride = true)) {
```

Comment:
Pleaaaase give my eye some `.` and `{ }` to hold on to 😆 

`onComplete(invokeAction(user, env map.(action.inherit(_)).getOrElse(action), payload, blocking, waitOverride = true))`

makes it imo easier to grasp what's going on.